Server/mods/deathmatch/logic/CScriptDebugging.cpp

```diff
@@ -143,16 +159,38 @@ void CScriptDebugging::PrintLog(const char* szText)
     }
 }
 
+bool CScriptDebugging::CheckForSufficientDebugLevel(std::uint8_t playerDebugLevel, std::uint8_t messageDebugLevel) const noexcept
+{
+    bool sufficientDebugLevel = false;
+
+    switch (static_cast<DebugMessageLevels>(messageDebugLevel))
+    {
+        case DebugMessageLevels::MESSAGE_TYPE_ERROR:
+            sufficientDebugLevel = (playerDebugLevel >= static_cast<std::uint8_t>(DebugScriptLevels::ERRORS_ONLY));
+            break;
+        case DebugMessageLevels::MESSAGE_TYPE_WARNING:
+            sufficientDebugLevel = (playerDebugLevel >= static_cast<std::uint8_t>(DebugScriptLevels::ERRORS_AND_WARNINGS));
+            break;
+        case DebugMessageLevels::MESSAGE_TYPE_INFO:
+        case DebugMessageLevels::MESSAGE_TYPE_CUSTOM:
+        case DebugMessageLevels::MESSAGE_TYPE_DEBUG:
+            sufficientDebugLevel = (playerDebugLevel == static_cast<std::uint8_t>(DebugScriptLevels::ALL));
+            break;
+    }
```

Comment:
Change
```cpp
bool CScriptDebugging::CheckForSufficientDebugLevel(std::uint8_t playerDebugLevel, std::uint8_t messageDebugLevel) const noexcept
```
to
```cpp
bool CScriptDebugging::CheckForSufficientDebugLevel(DebugMessageLevels playerDebugLevel, DebugMessageLevels messageDebugLevel) const noexcept
```
That way you can remove all static_asserts

Reply:
It wouldn't be so easy, it would require refactoring other parts of codebase, which would be out of scope for this PR. It was supposed to be tweak, not refactor. I also don't have will to do so - feel free to introduce any necessary changes afterwards. I've reverted it to default enum.